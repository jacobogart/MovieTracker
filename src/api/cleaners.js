import { imageUrl } from '../utils/API-logins';

export const cleanAllMovies = (moviesArray) => {
  const cleanMovies = moviesArray.map(movie => {
    const { 
      id, 
      title, 
      poster_path, 
      overview, 
      release_date, 
      vote_average 
    } = movie;

    return { 
      id, 
      title, 
      posterImg: `${imageUrl}${poster_path}`, 
      overview, 
      releaseDate: release_date, 
      rating: vote_average 
    };
  });

  return cleanMovies;
};

export const cleanPeople = (data) => {
  const director = data.crew.filter(person => person.job === "Director");
  const writer = data.crew.filter(person => person.job === "Writer");
  const cast = data.cast.map(person => {
    let { name, character, profile_path,  cast_id} = person;

    return { 
      name, 
      character, 
      headshot: `${imageUrl}${profile_path}`, 
      id: cast_id 
    };
  });

  return { director, writer, cast };
};