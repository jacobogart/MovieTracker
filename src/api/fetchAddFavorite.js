import { serverUrl } from "./pathNames";

export const fetchAddFavorite = (cardData) => {
  const url = `${serverUrl}/favorites/new`;
  const { 
    user, 
    title, 
    overview, 
    posterImg, 
    id, 
    rating, 
    releaseDate 
  } = cardData;
  let poster_path = posterImg.split('500')[1];
  const body = {
    movie_id: id, 
    user_id: user.id,
    title, 
    poster_path,
    release_date: releaseDate, 
    vote_average: rating, 
    overview
  };
  const init = {
    method: 'POST',
    body: JSON.stringify(body),
    headers: { 'Content-Type': 'application/json' }
  };

  return fetch(url, init)
    .then(response => {
      if (!response.ok) {
        throw Error('Failed to add favorite');
      } else {
        return response.json();
      }
    });
};
