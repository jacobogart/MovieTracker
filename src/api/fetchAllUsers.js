import { serverUrl } from "./pathNames";

export const fetchAllUsers = () => {
  return fetch(serverUrl)
    .then(response => {
      if (!response.ok) {
        throw new Error("Failed to create account");
      } else {
        return response.json();
      }
    })
    .then(data => data.data);
};
