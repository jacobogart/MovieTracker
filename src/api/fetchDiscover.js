import { jacobKey, baseUrl } from '../utils/API-logins'

const fetchDiscover = () => {
  return fetch(`${baseUrl}discover/movie${jacobKey}`)
    .then(response => {
      if (!response.ok) {
        throw new Error('Discover failed to fetch');
      } else {
        return response.json();
      }
  });
}

export default fetchDiscover;

