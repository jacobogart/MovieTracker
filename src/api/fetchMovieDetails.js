import { jacobKey } from '../utils/API-logins';
import { baseUrl } from './pathNames';

const fetchMoviePeople = (id) => {
  return fetch(`${baseUrl}movie/${id}/credits${jacobKey}`)
    .then(response => {
      if (!response.ok) {
        throw new Error('People failed to fetch');
      } else {
        return response.json();
      }
    });
};

export default fetchMoviePeople;
