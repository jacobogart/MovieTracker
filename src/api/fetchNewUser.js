import { serverUrl } from './pathNames';

export const fetchNewUser = (email, name, password) => {
  const body = JSON.stringify({ email, name, password });
  const init = {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body
  };
  const request = new Request(`${serverUrl}/new`, init);

  return fetch(request)
    .then(response => {
      if (!response.ok) {
        throw new Error('Failed to create account');
      } else {
        return response.json();
      }
    })
    .then(data => {
      return data.id;
    });
};