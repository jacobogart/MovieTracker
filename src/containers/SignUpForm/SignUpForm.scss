.signup-backdrop {
  @include transparency($theme-rgb);
  display: flex;
  justify-content: space-around;

  .SignUpForm {
    background: white;
    margin: 50px 0;
    padding: 50px;
    text-align: left;
    width: 80%;

    h2 {
      margin: 20px 0;
    }

    h4 {
      margin: 10px 0;
    }

    .flex-wrapper {
      display: flex;
      justify-content: space-between;

      form {
        @include flex-column;
        @include form-layout;
        flex: 1;
        margin: 30px 0;

        .cancel-link {
          margin: 0 10px;
        }
      }
    }
  }
}

.signup-blurb {
  @include box-shadow(#E5E5E5);
  height: 280px;
  margin: 40px 0 0 40px;
  position: relative;
  top: 18px;
  width: 300px;

  h3 {
    background: gray;
    margin: 0;
    padding: 20px;
  }

  .signup-sub-blurb {
    padding: 20px;
    min-width: 200px;

    ul {
      padding: 20px;
    }
  }
}