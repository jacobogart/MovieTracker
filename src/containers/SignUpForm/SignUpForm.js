import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { fetchNewUser } from '../../api/fetchNewUser';
import { connect } from 'react-redux';
import { updateUser } from '../../actions';
import { fetchAllUsers } from '../../api/fetchAllUsers';

export class SignUpForm extends Component {
  state = {
    email: '',
    name: '',
    password: '',
    passwordConfirm: '',
    error: ' '
  }

  handleChange = (e) => {
    const { name, value } = e.target;

    this.setState({
      [name]: value
    });
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const { email, name, password, passwordConfirm } = this.state;

    this.setState({ error: ' ' });

    fetchAllUsers()
      .then(data => {
        this.checkEmail(data, email);
        this.checkPasswords(password, passwordConfirm);
        if (this.state.error === ' ') {
          fetchNewUser(email, name, password)
            .then(id => {
              this.props.updateUser(id, name);
              this.props.history.push("/");
            })
            .catch(error => console.log(error));
        }
      })
      .catch(error => console.log(error));
  }

  checkEmail = (data, email) => {
    data.forEach(user => {
      if(user.email === email) {
        this.setState({ error: 'Email is already taken. Please try again.' });
      }
    });
  }

  checkPasswords = (password, passwordConfirm) => {
    if(password !== passwordConfirm) {
      this.setState({ error: 'Passwords do not match. Please try again.' });
    }
  }

  render() {
    let { error } = this.state;

    return (
      <div className="signup-backdrop">
        <div className="SignUpForm">
          <h2>Sign up for an account</h2>
          <h4>
            Sign up is free and easy! Fill out the form below to get
            started.
          </h4>
          <div className="flex-wrapper">
            <form
              onSubmit={this.handleSubmit}
              className="signup-form-inputs">
              <label htmlFor="signup-email-input">Email</label>
              <input
                type="text"
                id="signup-email-input"
                name="email"
                value={this.state.email}
                onChange={this.handleChange}
              />
              <label htmlFor="signup-name-input">Name</label>
              <input
                type="text"
                id="signup-name-input"
                name="name"
                value={this.state.name}
                onChange={this.handleChange}
              />
              <label htmlFor="signup-password-input">Password</label>
              <input
                type="password"
                id="signup-password-input"
                name="password"
                value={this.state.password}
                onChange={this.handleChange}
              />
              <label htmlFor="password-confirm-input">
                Confirm Password
              </label>
              <input
                type="password"
                id="password-confirm-input"
                name="passwordConfirm"
                value={this.state.passwordConfirm}
                onChange={this.handleChange}
              />
              <div className="error-holder">
                <p className="error-message">{error}</p>
              </div>
              <div className="signup-btn-wrapper">
                <button type="submit" className="signup-btn btn">
                  Sign Up
                </button>
                <Link to="/" className="cancel-link">
                  Cancel
                </Link>
              </div>
            </form>
            <article className="signup-blurb">
              <h3>Why Sign Up?</h3>
              <div className="signup-sub-blurb">
                <p>As a member you can:</p>
                <ul>
                  <li>View details of any movie</li>
                  <li>Keep track of favorite movies</li>
                  <li>Build and maintain an watchlist</li>
                </ul>
              </div>
            </article>
          </div>
        </div>
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  updateUser: (id, name) => dispatch(updateUser(id, name))
});

export default connect(null, mapDispatchToProps)(SignUpForm);

SignUpForm.propTypes = {
  history: PropTypes.object,
  location: PropTypes.object,
  match: PropTypes.object,
  updateUser: PropTypes.func
};


