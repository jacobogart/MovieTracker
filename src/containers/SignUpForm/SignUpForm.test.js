import React from "react";
import { shallow } from "enzyme";
import { SignUpForm, mapDispatchToProps } from "./SignUpForm";
import { fetchNewUser } from "../../api/fetchNewUser";

jest.mock("../../api/fetchNewUser");

describe("SignUpForm", () => {
  let wrapper, instance;
  let mockUpdateUser = jest.fn();
  fetchNewUser.mockImplementation(() => Promise.resolve(1));

  beforeEach(() => {
    wrapper = shallow(<SignUpForm updateUser={mockUpdateUser} />);
    instance = wrapper.instance();
  });

  afterEach(() => {
    fetchNewUser.mockClear();
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a defult state", () => {
    const mockDefaultState = {
      email: "",
      name: "",
      password: "",
      passwordConfirm: ""
    };
    wrapper = shallow(<SignUpForm />, { disableLifecycleMethods: true });
    expect(wrapper.state()).toEqual(mockDefaultState);
  });

  describe('handleChange', () => {
    const mockEvent = {
      target: {
        name: 'name',
        value: 'Jacob'
      }
    };

    it('should set the appropriate state to the input value', () => {
      instance.handleChange(mockEvent);
      expect(wrapper.state('name')).toEqual('Jacob');
    });
  });

  describe("handleSubmit", () => {
    const mockEvent = {
      preventDefault: () => {}
    }
    it("should invoke fetchNewUser", () => {
      instance.handleSubmit(mockEvent);
      expect(fetchNewUser).toHaveBeenCalled();
    });

    it("should invoke updateUser", () => {
      instance.handleSubmit(mockEvent);
      expect(mockUpdateUser).toHaveBeenCalled();
    });
  });
});