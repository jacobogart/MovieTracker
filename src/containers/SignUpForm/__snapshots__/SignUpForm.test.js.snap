// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm should match the snapshot 1`] = `
<div
  className="SignUpForm"
>
  <h2>
    Sign up for an account
  </h2>
  <p>
    Sign up is free and easy! Fill out the form below to get started.
  </p>
  <form
    className="signup-form-inputs"
    onSubmit={[Function]}
  >
    <label
      htmlFor="signup-email-input"
    >
      Email
    </label>
    <input
      id="signup-email-input"
      name="email"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="signup-name-input"
    >
      Name
    </label>
    <input
      id="signup-name-input"
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="signup-password-input"
    >
      Password
    </label>
    <input
      id="signup-password-input"
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
    <label
      htmlFor="password-confirm-input"
    >
      Confirm Password
    </label>
    <input
      id="password-confirm-input"
      name="passwordConfirm"
      onChange={[Function]}
      type="password"
      value=""
    />
    <div
      className="signup-btn-wrapper"
    >
      <button
        className="signup-btn btn"
        type="submit"
      >
        Sign Up
      </button>
      <Link
        className="cancel-link"
        to="/"
      >
        Cancel
      </Link>
    </div>
  </form>
  <article
    className="signup-blurb"
  >
    <h3>
      Why Sign Up?
    </h3>
    <div
      className="signup-sub-blurb"
    >
      <p>
        As a member you can:
      </p>
      <ul>
        <li>
          Log movies you have watched
        </li>
        <li>
          Keep track of favorite movies
        </li>
        <li>
          Build and maintain an watchlist
        </li>
      </ul>
    </div>
  </article>
</div>
`;
